specification/dartLangSpec.tex

```diff
@@ -1267,8 +1267,10 @@ \section{Errors and Warnings}
 are situations that do not preclude execution,
 but which are unlikely to be intended,
 and likely to cause bugs or inconveniences.
-A static warning must be reported by a Dart compiler
-before the associated code is executed.
+A Dart compiler is free to report some, all,
+or none of the specified static warnings before executing the associated code.
```

Comment:
Nit: Compilers don't execute code, so maybe "before the associated code is executed."

(Which would even allow the compiler to insert code that prints a warnings when the compiled program is run.)

We can then wonder why it's not allowed to report the warning *after* the code is executed, but now I think I'm just being pointlessly pedantic. :grin:

Reply:
Good point. Done: https://github.com/dart-lang/language/pull/3584